### PR TITLE
Fixed noauth/normal auth helper handling

### PR DESCRIPTION
**Pull request by kunagpal**
This ensures collection run sanity in cases where inherited auth helpers are involved.

### CODE REVIEW

File: lib/converters/v2.0.0/converter-v2-to-v1.js [hunk 1]
```diff
@@ -368,10 +368,10 @@ _.assign(Builders.prototype, {
             name: item.name,
             collectionId: collectionId,
             method: item.request ? item.request.method : undefined,
-            currentHelper: currentHelper,
-            helperAttributes: helperAttributes
+            currentHelper: currentHelper
         };
 
+        helperAttributes && (request.helperAttributes = helperAttributes);
```
Comment (kamalaknn) [inline lib/converters/v2.0.0/converter-v2-to-v1.js]: `request.helperAttributes = helperAttributes` would set it to `undefined`, which will be ignored by the receiver right?

Comment (kunagpal) [reply lib/converters/v2.0.0/converter-v2-to-v1.js]: Yes. In addition, the truthy-ness of `helperAttributes` is being checked first.

File: lib/normalizers/v1.js [hunk 2]
```diff
@@ -42,16 +42,14 @@ _.assign(Builders.prototype, {
     auth: function (entityV1) {
         if (!entityV1) { return; }
         if (util.notLegacy(entityV1, 'auth') && entityV1.auth) {
-            if (entityV1.auth.type === 'noauth') { return; }
-
-            return util.sanitizeAuthArray(entityV1);
+            return (entityV1.auth.type === 'noauth') ? entityV1.auth : util.sanitizeAuthArray(entityV1);
```
Comment (kamalaknn) [inline lib/normalizers/v1.js]: Can we return `{ type: 'noauth' }`, so any other properties in `auth` are removed?

File: lib/normalizers/v1.js [hunk 3]
```diff
@@ -42,16 +42,14 @@ _.assign(Builders.prototype, {
...
         }
-        if (!entityV1.currentHelper || (entityV1.currentHelper === 'normal')) { return; }
+        if (!entityV1.currentHelper) { return; }
```
Comment (kamalaknn) [inline lib/normalizers/v1.js]: Same as above

Comment (kunagpal) [reply lib/normalizers/v1.js]: This is the case where the auth properties are absent altogether, indicating that the auth for this entity can be overridden.

**Pull request merged**